cmd/curio/market.go

```diff
@@ -1,122 +1,370 @@
-		marketRPCInfoCmd,
 		marketSealCmd,
+		marketImportdataCmd,
+		marketAddOfflineURLCmd,
 	},
 }
 
-var marketRPCInfoCmd = &cli.Command{
+var marketSealCmd = &cli.Command{
+	Name:  "seal",
+	Usage: "start sealing a deal sector early",
 	Flags: []cli.Flag{
-		&cli.StringSliceFlag{
-			Name:  "layers",
-			Usage: "list of layers to be interpreted (atop defaults). Default: base",
+		&cli.StringFlag{
+			Name:     "actor",
+			Usage:    "Specify actor address to start sealing sectors for",
+			Required: true,
+		},
+		&cli.BoolFlag{
+			Name:  "synthetic",
+			Usage: "Use synthetic PoRep",
+			Value: false,
 		},
 	},
+	ArgsUsage: "<sector>",
 	Action: func(cctx *cli.Context) error {
-		db, err := deps.MakeDB(cctx)
+		act, err := address.NewFromString(cctx.String("actor"))
+		if err != nil {
+			return xerrors.Errorf("parsing --actor: %w", err)
+		}
+
+		if cctx.Args().Len() > 1 {
+			return xerrors.Errorf("specify only one sector")
+		}
+
+		sec := cctx.Args().First()
+
+		sector, err := strconv.ParseUint(sec, 10, 64)
+		if err != nil {
+			return xerrors.Errorf("failed to parse the sector number: %w", err)
+		}
+
+		ctx := reqcontext.ReqContext(cctx)
+		dep, err := deps.GetDepsCLI(ctx, cctx)
 		if err != nil {
 			return err
 		}
 
-		layers := cctx.StringSlice("layers")
+		return storageIngest.SealNow(ctx, dep.Chain, dep.DB, act, abi.SectorNumber(sector), cctx.Bool("synthetic"))
+	},
+}
+
+var marketImportdataCmd = &cli.Command{
+	Name:  "import-data",
+	Usage: "Import data for offline deal",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:     "actor",
+			Usage:    "Specify actor address to start sealing sectors for",
+			Required: true,
+		},
+	},
+	ArgsUsage: "<deal UUID> <file> <host:port>",
+	Action: func(cctx *cli.Context) error {
+		if cctx.Args().Len() != 3 {
+			return xerrors.Errorf("incorrect number of arguments")
+		}
+
+		idStr := cctx.Args().First()
 
-		cfg, err := deps.GetConfig(cctx.Context, layers, db)
+		id, err := uuid.Parse(idStr)
 		if err != nil {
-			return xerrors.Errorf("get config: %w", err)
+			return err
 		}
 
-		ts, err := lmrpc.MakeTokens(cfg)
+		fileStr := cctx.Args().Get(1)
+		fpath, err := homedir.Expand(fileStr)
 		if err != nil {
-			return xerrors.Errorf("make tokens: %w", err)
+			return err
 		}
 
-		var addrTokens []struct {
-			Address string
-			Token   string
+		f, err := os.Open(fpath)
+		if err != nil {
+			return err
 		}
 
-		for address, s := range ts {
-			addrTokens = append(addrTokens, struct {
-				Address string
-				Token   string
-			}{
-				Address: address.String(),
-				Token:   s,
-			})
+		defer func() {
+			_ = f.Close()
+		}()
+
+		st, err := f.Stat()
+		if err != nil {
+			return err
 		}
 
-		sort.Slice(addrTokens, func(i, j int) bool {
-			return addrTokens[i].Address < addrTokens[j].Address
-		})
+		rawSize := st.Size()
 
-		for _, at := range addrTokens {
-			fmt.Printf("[lotus-miner/boost compatible] %s %s\n", at.Address, at.Token)
+		fUrl := "file:///" + fpath
```

Comment:
CanAccept for piecePark has been modified to check for this. If the task has file:/// scheme then match the host:port. If same as DB then accept the task otherwise don't take the task.